feature/issue-10: Fixed bug where variable dimensions are assumed to be the same across all variables in a group

nlenssen2013 opened the pull request:
Github Issue: #10 

### Description

Separated variable dimensions for each variable rather than the entire group

### Overview of verification done

test written checking that each variable dimensions are the same as the one that is being created in recombine_groups method

### Overview of integration done

NA

## PR checklist:

* [ x] Linted
* [ x] Updated unit tests
* [ x] Updated changelog
* [ ] Integration testing

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

frankinspace commented:
Couple things here:

In the PR description, remove the italicized text and replace it with information; the italicized text is just meant to be a prompt to help you write the PR.

The changelog only gets 1 entry per issue and all changes should be added to the Unreleased section. When a release is created, the Unreleased section gets named with the release version and a new empty unreleased section is created.

If the issue is fixing a bug add a line under fixed. 
If it's a new feature (as in a new capability that the service did not previously offer) that goes in added. 
Removed is for when a feature or capability is being removed... typically used for when there are breaking changes to the interfaces used by clients.
Deprecated is similar to removed except it is to be used for allowing clients time to migrate off of the function being deprecated.
Security is for patching security holes.

Also, make sure to pull in from upstream develop as I have updated the changelog there.

nlenssen2013 commented:
> 

Got it. I did pull down the updated changelog from upstream develop, and those changes are included this branch. Still trying to get the workflow down. So when upstream develop is changed and I've made a branch locally off of the outdated upstream develop, what do I do to update the branch I'm working on? git pull upstream/develop and then git merge feature/issue--10?

frankinspace commented:
Usually what I do is:
```
# Update all remote refs
git fetch --all 
# Make sure I'm on the branch I want to pull changes into
git checkout feature/issue--10
# Pull in using a merge commit from the 'upstream' remote
git pull upstream develop
```
This assumes you've added `git@github.com:podaac/l2ss-py.git` as a remote named `upstream`

The `git pull` might result in merge conflicts if the same file has been modified on the remote and on your local branch. In that situation you need to resolve the conflicts and `git add` the files to mark them as resolved. Once all conflicts are resolved you can `git commit` to finish the merge then `git push origin feature/issue--10` to push the merge to your fork (assuming you've named your forked remote `origin`)

nlenssen2013 commented:
> Usually what I do is:
> 
> ```
> # Update all remote refs
> git fetch --all 
> # Make sure I'm on the branch I want to pull changes into
> git checkout feature/issue--10
> # Pull in using a merge commit from the 'upstream' remote
> git pull upstream develop
> ```
> 
> This assumes you've added `git@github.com:podaac/l2ss-py.git` as a remote named `upstream`
> 
> The `git pull` might result in merge conflicts if the same file has been modified on the remote and on your local branch. In that situation you need to resolve the conflicts and `git add` the files to mark them as resolved. Once all conflicts are resolved you can `git commit` to finish the merge then `git push origin feature/issue--10` to push the merge to your fork (assuming you've named your forked remote `origin`)

Great, pretty sure that worked

skorper reviewed (commented):
I don't think the merge went right because the changelog looks a bit unusual, there are two `[Unreleased]` sections and some random characters I don't are intentional

nlenssen2013 commented:
> I don't think the merge went right because the changelog looks a bit unusual, there are two `[Unreleased]` sections and some random characters I don't are intentional

Updated the released number and removed the characters

frankinspace commented on podaac/subsetter/subset.py:
```diff
@@ -918,8 +918,8 @@ def get_nested_group(dataset, group_path):
             var_group = get_nested_group(base_dataset, var_name)
-            var_dims = list(var_group.dimensions.keys())
             variable = dataset.variables[var_name]
+            var_dims = [x.split('__')[-1] for x in dataset.variables[var_name].dims]
```
```suggestion
            var_dims = [x.split(GROUP_DELIM)[-1] for x in dataset.variables[var_name].dims]
```

frankinspace reviewed (approved):
I fixed the changelog. Just need the one line that links the issue and gives a brief summary of the bug and fix. Code changes can all be view via the link to the issue and from there the PR itself.
